src/System.Runtime/tests/System/Lazy.cs

```diff
@@ -82,98 +126,95 @@ public static void TestValueExceptions()
 
                 lazilyAllocatedValue = lazy.Value;
             });
-
-            Assert.Throws<MissingMemberException>(() =>
-            {
-                lazy = new Lazy<string>();
-                lazilyAllocatedValue = lazy.Value;
-            });
         }
 
         [Fact]
-        public static void TestValueFactoryExceptions()
+        public static void TestValue_ThrownException_DoesntCreateValue()
         {
-            Lazy<string> l = new Lazy<string>(() =>
-                {
-                    int zero = 0;
-                    int x = 1 / zero;
-                    return "";
-                }, true);
-            string s;
+            var lazy = new Lazy<string>(() =>
+            {
+                int zero = 0;
+                int x = 1 / zero;
+                return "";
+            }, true);
 
-            Assert.Throws<DivideByZeroException>(() => s = l.Value);
-            Assert.Throws<DivideByZeroException>(() => s = l.Value);
+            Assert.Throws<DivideByZeroException>(() => lazy.Value);
+            Assert.Throws<DivideByZeroException>(() => lazy.Value);
```

Comment:
Then, if there isn't one already, there should be a similar test that uses PublicationOnly and that uses Assert.NotSame to verify they're not the same instance.